src/syft/core/node/domain/domain.py

```diff
@@ -151,56 +153,114 @@ def get_request_status(self, message_request_id: UID) -> RequestStatus:
         # must have been rejected
         return RequestStatus.Rejected
 
+    @syft_decorator(typechecking=True)
+    def _get_object(self, request: RequestMessage) -> Optional[Any]:
+        try:
+            obj_msg = GetObjectAction(
+                id_at_location=request.object_id,
+                address=request.owner_address,
+                reply_to=self.address,
+                delete_obj=False,
+            )
+
+            service = self.immediate_msg_with_reply_router[type(obj_msg)]
+            response = service.process(
+                node=self, msg=obj_msg, verify_key=self.root_verify_key
+            )
+            if response:
+                obj = getattr(response, "obj", None)
+                if obj is not None:
+                    return obj
+        except Exception as e:
+            logger.critical(f"Exception getting object for {request}. {e}")
+        return None
+
+    def _count_elements(self, obj: object) -> int:
+        elements = 1
+
+        nelement = getattr(obj, "nelement", None)
+        if nelement is not None:
+            elements = max(elements, int(nelement()))
+
+        length = getattr(obj, "__len__", None)
+        if length is not None:
+            elements = max(elements, int(length()))
+
+        return elements
+
+    def _accept(self, request: RequestMessage) -> None:
+        logger.debug(f"Calling accept on request: {request.id}")
+        request.destination_node_if_available = self
+        request.accept()
+
+    def _deny(self, request: RequestMessage) -> None:
+        logger.debug(f"Calling deny on request: {request.id}")
+        request.destination_node_if_available = self
+        request.deny()
+
+    def _try_deduct_quota(
+        self, handler: Dict[Union[str, String], Any], obj: Any
+    ) -> bool:
+        if "action" in handler and handler["action"] == "accept":
+            element_count = self._count_elements(obj=obj)
+            result = handler["element_quota"] - element_count
+            if result >= 0:
+                # the request will be accepted so lets decrement the quota
```

Comment:
Q: What represents the quota for an element?

Reply:
It is the total number of "elements" you can request from a handler rule, so you could limit it to 100 and get no requests automatically approved which exceed 100 nelements or len(), which are cumulative between requests. Its not perfect since a nested List would not work, but for Tensors which are the main concern, the nelement should work.